mizar/daemon/interface_service.py

```diff
@@ -203,12 +203,20 @@ def ConsumeInterfaces(self, request, context):
         logger.info("Consuming interfaces for pod: {} Current Queue: {}".format(
             requested_pod_name, list(self.pod_dict)))
 
-        if self.pod_dict:
-            if requested_pod_name in self.pod_dict:
-                if self.pod_dict[requested_pod_name]:
-                    # Interfaces for the Pod has been produced
-                    self.pod_dict.pop(requested_pod_name)
-                    return self._ConsumeInterfaces(requested_pod_name, request)
+        # The success of this function depends on ProduceInterfaces which is executed in another process.
+        # So using while here to wait until ProduceInterfaces has been done.
+        start = time.time()
+        while True:
+            if self.pod_dict:
+                if requested_pod_name in self.pod_dict:
+                    if self.pod_dict[requested_pod_name]:
+                        # Interfaces for the Pod has been produced
+                        self.pod_dict.pop(requested_pod_name)
+                        return self._ConsumeInterfaces(requested_pod_name, request)
+            time.sleep(0.5)
```

Comment:
Can we please make the 0.5 a constant? (in the same file next to CONSUME_INTERFACE_TIMEOUT is good enough)

Reply:
Const has been added.